### PR TITLE
fix: smolagents plugin nullable params

### PR DESCRIPTION
**Pull request by Prat011**
> [!IMPORTANT]
> In `toolset.py`, `_wrap_tool()` now sets `nullable: True` for non-required parameters.
> 
>   - **Behavior**:
>     - In `toolset.py`, `_wrap_tool()` now sets `nullable: True` for non-required parameters in `params`.
>   - **Misc**:
>     - No other files or functions are affected.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ComposioHQ%2Fcomposio&utm_source=github&utm_medium=referral)<sup> for b10966f868132577eeebe58786e775dd3a9613ca. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: python/plugins/smolagent/composio_smol/toolset.py [hunk 1]
```diff
@@ -121,6 +121,10 @@ def _wrap_tool(
         )
         # Flatten and format the parameters structure
         params = schema_params["properties"]
```
Comment (shreysingla11) [inline python/plugins/smolagent/composio_smol/toolset.py:123]: Consider adding error handling for cases where `params` might be None or not a dictionary. Also, it would be good to validate that `param_info` is a mutable dictionary before attempting to modify it. Something like:
```python
if not isinstance(params, dict):
    raise ValueError("Schema parameters properties must be a dictionary")
```

File: python/plugins/smolagent/composio_smol/toolset.py [hunk 2]
```diff
@@ -121,6 +121,10 @@ def _wrap_tool(
...
+        required_params = schema_params.get("required", [])
```
Comment (shreysingla11) [inline python/plugins/smolagent/composio_smol/toolset.py:124]: Consider adding a docstring or comment explaining the purpose of the nullable flag and how it affects parameter handling. This will help future maintainers understand the code's intention better.

**Comment by shreysingla11** [comment]
### Code Review Summary

The changes look generally good and implement a useful feature for handling optional parameters in the tool schema. Here's a breakdown of the review:

#### Positive Aspects ✅
- Clean implementation of nullable flag for optional parameters
- Efficient code with O(n) complexity
- No breaking changes to existing functionality
- Follows the existing code structure

#### Suggestions for Improvement 🔧
1. Add error handling for malformed schemas
2. Add documentation explaining the nullable flag's purpose
3. Consider adding unit tests for:
   - Optional parameters being marked as nullable
   - Required parameters not being marked as nullable
   - Edge cases with malformed schemas

#### Code Quality Rating: 7/10
Good functionality but could benefit from additional error handling and documentation.

The core changes are solid and the PR can be merged after addressing the suggested improvements.

**Review by angrybayblade — APPROVED** [review]
